### DISCUSSION TITLE
Correct way to transform field data for CRUD create/update?

### DISCUSSION THREAD

**Discussion by alancwoo** [Laravel-Backpack/community-forum#536] [Q&A (Help)] [answered]
Let's say I have a model `Issue` which has a `year` and `month` field.

If I want to use the [month field](https://backpackforlaravel.com/docs/5.x/crud-fields#month-1), what is the correct way to transform this data so that it properly splits this and sets the `year` and `month` accordingly?

I can get this to work for creation by defining a `create` function on the model:

```
public function create($data)
{
  [$year, $month] = explode('-', $data['month']);
  $this->year = $year;
  $this->month = $month;
  $this->save();
  return $this;
}
```

but of course this doesn't work when `editing` as the month field is not correctly set:

```
CRUD::field('month')
            ->type('month')
```

I'm wondering what is the correct way to transform data in the CrudController? Or would the better way be to adjust the database to hold the `YYYY-MM` string? (preferably not as I would like to be able to query years/months directly)

**Comment by christmex** [comment, marked as answer]
Hi mate @alancwoo , try this one

so we have ``` Issue ``` model, create a laravel accessor inside your ``` Issue ``` model, you can add this function
```php
public function getCleanYearAttribute()
{ 
    return explode('-',$this->year)[0];
}

protected function year(): Attribute
{
    return Attribute::make(
        get: fn (string $value) => $value.'-'.$this->month,
    );
}
```

and your ```IssueCrudController``` should be like this
```php
//this is for setupListOperation
CRUD::column('clean_year'); //change year to clean_year

//this is for setupCreateOperation
CRUD::field('year')->type('month')->on('saving', function ($entry) {
    $explodeData = explode('-', $entry->year);
    $entry->year = $explodeData[0];
    $entry->month = $explodeData[1];
});
```
Reference
[laravel accessor](https://laravel.com/docs/10.x/eloquent-mutators#defining-an-accessor)
[Backpack model events](https://backpackforlaravel.com/docs/5.x/release-notes#use-model-events-inside-crudcontrollers)

for now, that's from me, let me know if you find any issues, ill try to find another way and update it here, so we can improve it. Cheer

**Reply by alancwoo** [reply]
Thank you!
